### PR TITLE
NxGlobalSidebar example page RSC-489

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-489

### CODE REVIEW

**Comment by rpokorny** [comment]
You added two different logo files.  Are they both used?

**Comment by rpokorny** [comment]
Needs a page for `NxStatefulGlobalSidebar`

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx:20]: This should be pulled in via a `require` call

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.scss
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.scss:7]: This is kinda quick and dirty.  We should probably have less sloppy code in an example

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:39]: This isn't what I ended up naming it in my branch.  We should probably meet with @dturner32 and decide what the best name is

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:42]:
```suggestion
            Additional custom content may be added below the navigation links if desired, care should be taken that such
```

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:19]: Is there a reason this isn't using `GalleryDescriptionTile`?

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:63]: No details?

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:80]: I find the descriptions on these two confusing

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:88]:
```suggestion
                    The path to an image placed in the top left corner which displays branding and product name.
```

Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:88]: I believe there are requirements on the size and contents of the image which should be documented here.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:191]: Keep in mind pretty much all of the RSC styles you're using in this file have convenience components.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx:42]: I don't think it's a good idea to reference the gallery's internal components within an example

File: gallery/src/handleQueryParams.ts
Comment (rpokorny) [inline gallery/src/handleQueryParams.ts:31]: What's the purpose of this?  We have some pages that have the global sidebar and some that don't, so to additionally be able to hide it on pages that have it seems redundant

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:209]: I'm not sure this example is necessary here.  I have examples of this in RSC-487 where it was decided whether the old sidebar is supposed to appear next to or underneath of the new header.

**Comment by glenhunter** [comment]
> You added two different logo files. Are they both used?

Yes. One is a `png` and the other an `svg`. Not that the format matter especially. Also one has text, the other doesn't. 

Though if you're saying (as below) that I don't need both examples I'll likely get rid of one logo.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:39]: > This isn't what I ended up naming it in my branch. We should probably meet with @dturner32 and decide what the best name is

I was also wondering if it should be pulled until your work is merged.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx:42]: > I don't think it's a good idea to reference the gallery's internal components within an example

Ok, that's NBD to change, but why?

File: gallery/src/handleQueryParams.ts
Comment (glenhunter) [reply gallery/src/handleQueryParams.ts]: > What's the purpose of this? We have some pages that have the global sidebar and some that don't, so to additionally be able to hide it on pages that have it seems redundant

It's old and should have been removed: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/2e9380a7be441078ceb31f7af2a5269f3bca2c2b

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > I'm not sure this example is necessary here. I have examples of this in RSC-487 where it was decided whether the old sidebar is supposed to appear next to or underneath of the new header.

Removed here: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/58ebce60573ddfb22d395e8a8ebca0557f73596d

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > Keep in mind pretty much all of the RSC styles you're using in this file have convenience components.

I've converted most if not all here: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/58ebce60573ddfb22d395e8a8ebca0557f73596d

For me the more complex convenience components (NxTile et.al.) are not more convenient than writing the straight HTML. As far as I'm concerned we could have stopped with the basic HTML tags. I am not in favour of replacing HTML with convenience components and would have pushed back hard were I aware that was your line of thinking.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:38]: > I was also wondering if it should be pulled until your work is merged.

I'm fine with that

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx
Comment (rpokorny) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx]: > Ok, that's NBD to change, but why?

Because it will get people off track trying to figure out what it is and potentially how it works

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > Copy/pasted, not the right header for this section.

Fixed here: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/190afc675e47acd0ce72dab365dd5d16e035f5ec

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > I find the descriptions on these two confusing

Hopefully this is clearer: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/190afc675e47acd0ce72dab365dd5d16e035f5ec

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > No details?

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/190afc675e47acd0ce72dab365dd5d16e035f5ec

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx]: > This should be pulled in via a `require` call

Ok: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/10799b9adea5f60d3454929b3771140cc3d62dd8

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > Is there a reason this isn't using `GalleryDescriptionTile`?

No reason that I can remember: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/684ad398f518d2d3bfc1bbc3cc36ebf2d3abefd4

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > I believe there are requirements on the size and contents of the image which should be documented here.

I cannot find any requirements in the mockups. Derrek?

Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: @dturner32 ^^

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarWithPageSidebarExample.tsx]: > I don't think it's a good idea to reference the gallery's internal components within an example

This is removed now that the extra sidebar example is gone

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.scss
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.scss]: > This is kinda quick and dirty. We should probably have less sloppy code in an example

It was meant to be quick and dirty. I guess it was too quick, too dirty. Cleaner version here: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/50db41402d0baec8706a47d1345d6958a1669925

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:57]: It's not whether it starts open or closed, it's whether it is open or closed.  This being the stateless version of the component.

Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > It's not whether it starts open or closed, it's whether it is open or closed. This being the stateless version of the component.

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/8302c5865720b50cc8af378bb27c9e3882751783

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > > I was also wondering if it should be pulled until your work is merged.
> 
> I'm fine with that

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/ad49c6917bc88d132ffe8a9b98b717d7fb4fb80a

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > I cannot find any requirements in the mockups. Derrek?

I was referring to the aspect ratio and the fact that the image must include the text name of the product. This was discussed during the implementation ticket for the sidebar I believe.

Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > > I cannot find any requirements in the mockups. Derrek?
> 
> I was referring to the aspect ratio and the fact that the image must include the text name of the product. This was discussed during the implementation ticket for the sidebar I believe.

That's fine, but it should still be documented in the mockups, especially the aspect ratio.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:82]: Minor: I typically use alerts for this kind of callout

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarExample.tsx
Comment (rpokorny) [inline gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarExample.tsx:21]: Needs `.nx-page-content--full-width`

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:30]: Should mention that it should only be used with section scrolling and full-width mode

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:82]: > Minor: I typically use alerts for this kind of callout

Or if you do keep this the way it is, there's a whitespace issue around the `<strong>` that at least needs to be fixed.

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx:116]: These aren't things

File: gallery/visualtests/nxGlobalSidebar.js
Comment (rpokorny) [inline gallery/visualtests/nxGlobalSidebar.js:10]: This should be in a `beforeEach` block.  The way you have it, the 'looks right when closed' test will only work if the 'looks right' test runs right before it.  Tests should be independent of one another.

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarExample.tsx:16]:
```suggestion
    <div className="nx-page-content nx-page-content--full-width">
```

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:30]: > Should mention that it should only be used with section scrolling and full-width mode

Better docs here: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/5f3c9d3f3313a84fe9cb7f904e3d0b504d50393c

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarExample.tsx
Comment (glenhunter) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarExample.tsx]: > Needs `.nx-page-content--full-width`

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/64548b0220c92b417c896a5c6a065ac3c816eb4b

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx]: > Minor: I typically use alerts for this kind of callout

The alert is better: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/3a85d6147904a438c5742677e401a7c9797ec515

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx]: > These aren't things

Got a little exuberant with my search and replace: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/3a85d6147904a438c5742677e401a7c9797ec515

**Comment by rpokorny** [comment]
Console errors:
image

File: gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx
Comment (rpokorny) [inline gallery/src/components/NxGlobalSidebar/NxGlobalSidebarPage.tsx:32]:
```suggestion
          <NxCode>.nx-html--page-scrolling</NxCode> is <strong>not</strong> used) and in full-width mode (
```

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx
Comment (rpokorny) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx]: > Got a little exuberant with my search and replace: 3a85d61

FWIW I'm not sure duplicating all of the documentation is a good idea, as it means we'll have to remember to make any changes in both places.  I'd shorten the stateful page docs to defer to the stateless ones for most things.

File: gallery/visualtests/nxGlobalSidebar.js
Comment (rpokorny) [reply gallery/visualtests/nxGlobalSidebar.js:9]: > Ok, I didn't find the uses of beforeEach in other tests to be very clear so I hope that this is correct: 3c3645a

That's not wrong but those nested `describe` blocks are a bit redundant.  I'd just have the outer `describe`, the `beforeEach`, and then two tests: "it looks right when open" and "it looks right when closed".

**Comment by glenhunter** [comment]
> Console errors:
> image

I reused the `aria-label` since it was there, it was correct, and I think that the tooltip matching the aria-label makes sense.

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/d1e9607cc8384cf31982fa97b8da5123abe04745

**Comment by glenhunter** [comment]
> > I reused the aria-label since it was there, it was correct, and I think that the tooltip matching the aria-label makes sense.
> > [d1e9607](https://github.com/sonatype/sonatype-react-shared-components/commit/d1e9607cc8384cf31982fa97b8da5123abe04745)
> 
> The `aria-label` is now redundant. `NxTooltip` manages the accessible name of the button

Ok: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/1c37b114f2c520db09eb49639d2a913033c6b3f6

File: gallery/visualtests/nxGlobalSidebar.js
Comment (glenhunter) [reply gallery/visualtests/nxGlobalSidebar.js:9]: > > Ok, I didn't find the uses of beforeEach in other tests to be very clear so I hope that this is correct: [3c3645a](https://github.com/sonatype/sonatype-react-shared-components/commit/3c3645a17ad72b177f93f9bf1b21925e196da596)
> 
> That's not wrong but those nested `describe` blocks are a bit redundant. I'd just have the outer `describe`, the `beforeEach`, and then two tests: "it looks right when open" and "it looks right when closed".

Ok: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/1f6450fe74f09d7308f0f02c977ec8caab236669

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx]: > > Got a little exuberant with my search and replace: [3a85d61](https://github.com/sonatype/sonatype-react-shared-components/commit/3a85d6147904a438c5742677e401a7c9797ec515)
> 
> FWIW I'm not sure duplicating all of the documentation is a good idea, as it means we'll have to remember to make any changes in both places. I'd shorten the stateful page docs to defer to the stateless ones for most things.

Ah, yes good point: https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/6cd6ba526d753b12f7b0f2d21f9f0a8d2bb8a794

Comment (rpokorny) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx]: > Ah, yes good point: 6cd6ba5

Might want to make clear that it has `defaultOpen` _instead of_ `isOpen`

**Review by rpokorny — APPROVED** [review]

File: gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx
Comment (glenhunter) [reply gallery/src/components/NxStatefulGlobalSidebar/NxStatefulGlobalSidebarPage.tsx]: > > Ah, yes good point: [6cd6ba5](https://github.com/sonatype/sonatype-react-shared-components/commit/6cd6ba526d753b12f7b0f2d21f9f0a8d2bb8a794)
> 
> Might want to make clear that it has `defaultOpen` _instead of_ `isOpen`

https://github.com/sonatype/sonatype-react-shared-components/pull/371/commits/d45e57ab9e169b3c9d39e5c832364c74aca4bc25

**Pull request merged**
